cms/static/js/views/settings/main.js

```diff
@@ -146,7 +162,36 @@ var DetailsView = ValidatingView.extend({
         'course_image_asset_path': 'course-image-url',
         'pre_requisite_courses': 'pre-requisite-course',
         'entrance_exam_enabled': 'entrance-exam-enabled',
-        'entrance_exam_minimum_score_pct': 'entrance-exam-minimum-score-pct'
+        'entrance_exam_minimum_score_pct': 'entrance-exam-minimum-score-pct',
+        'course_settings_learning_fields': 'course-settings-learning-fields',
+        'add_course_learning_info': 'add-course-learning-info',
+        'add_course_instructor_info': 'add-course-instructor-info',
+        'course_learning_info': 'course-learning-info'
+    },
+
+    addLearningFields: function() {
+        /*
+        * Add new course learning fields.
+        * */
+        var existing_info = _.clone(this.model.get('learning_info')),
+            info = '';
+        existing_info[existing_info.length] = info;
+        this.model.set('learning_info', existing_info);
+        this.learning_info_view.render();
+    },
+
+    addInstructorFields: function() {
+        /*
+        * Add new course instructor fields.
+        * */
+        var existing_info = _.clone(this.model.get('instructor_info')),
+            index = existing_info.instructors.length,
+            data = JSON.parse('{"name": "","title": "","organization": "","image": "","bio": ""}');
```

Comment:
Nit: it might have been cleaner to use a model that represents these fields, and then having the view use a collection.